picketlink/src/main/java/org/wildfly/extension/picketlink/idm/model/DefaultRemoveStepHandler.java

```diff
@@ -22,52 +22,20 @@
 
 package org.wildfly.extension.picketlink.idm.model;
 
-import org.jboss.as.controller.OperationContext;
-import org.jboss.as.controller.OperationFailedException;
-import org.jboss.as.controller.PathAddress;
-import org.jboss.as.controller.RestartParentResourceRemoveHandler;
-import org.jboss.as.controller.registry.Resource;
-import org.jboss.dmr.ModelNode;
-import org.jboss.msc.service.ServiceName;
-import org.wildfly.extension.picketlink.common.model.ModelElement;
-import org.wildfly.extension.picketlink.idm.service.PartitionManagerService;
-
-import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP_ADDR;
+import org.jboss.as.controller.AbstractRemoveStepHandler;
 
 /**
  * <p>This remove handler is used during the removal of all partition-manager child resources. Its purpose is restart the
  * identity store services prior to the child removal, so we can stop all store services properly before restarting the parent.</p>
  *
  * @author Pedro Silva
  */
-public class DefaultRemoveStepHandler extends RestartParentResourceRemoveHandler {
+public class DefaultRemoveStepHandler extends AbstractRemoveStepHandler {
 
     static final DefaultRemoveStepHandler INSTANCE = new DefaultRemoveStepHandler();
 
     private DefaultRemoveStepHandler() {
-        super(ModelElement.PARTITION_MANAGER.getName());
-    }
-
-    @Override
-    protected void updateModel(OperationContext context, ModelNode operation) throws OperationFailedException {
-        super.updateModel(context, operation);
-
-        PathAddress partitionManagerAddress = getParentAddress(PathAddress.pathAddress(operation.require(OP_ADDR)));
-        Resource partitionManagerResource = context.readResourceFromRoot(partitionManagerAddress);
-        ModelNode parentModel = Resource.Tools.readModel(partitionManagerResource);
-
-        PartitionManagerAddHandler.INSTANCE.validateModel(context, partitionManagerAddress.getLastElement().getValue(), parentModel);
```

Comment:
Not really, PartitionManagerAddHandler.INSTANCE.validateModel() despite its name mostly does service creation.
I've extracted the minor actual validation part and left it in, so that minor validation that is going on will still continue to work as before.